src/ConnectedMode.UnitTests/StatefulServerBranchProviderTests.cs

```diff
@@ -46,20 +47,28 @@ public async Task GetServerBranchNameAsync_WhenCalled_UsesCache()
             var serverBranchProvider = CreateServerBranchProvider("Branch");
             var activeSolutionBoundTracker = Mock.Of<IActiveSolutionBoundTracker>();
 
-            var testSubject = CreateTestSubject(serverBranchProvider.Object, activeSolutionBoundTracker);
+            var threadHandling = new Mock<IThreadHandling>();
+            threadHandling.Setup(x => x.RunOnBackgroundThread(It.IsAny<Func<Task<string>>>()))
+                .Returns<Func<Task<string>>>(x => x()); // passthrough - call whatever was passed
```

Comment:
We need this so that (1) we can check that the thread handling mock was called as expected, and (2) the inner method is also executed, so we can check that the mapper mock was called.